pkg/virt-launcher/virtwrap/device/hostdevice/sriov/pcipool_netstatus.go

```diff
@@ -32,10 +34,17 @@ type PCIAddressWithNetworkStatusPool struct {
 func NewPCIAddressPoolWithNetworkStatus(networkPCIMapBytes []byte) (*PCIAddressWithNetworkStatusPool, error) {
 	pool := &PCIAddressWithNetworkStatusPool{}
 
-	var networkPciMap map[string]string
-	err := json.Unmarshal(networkPCIMapBytes, &networkPciMap)
+	var networkInfo downwardapi.NetworkInfo
+	err := json.Unmarshal(networkPCIMapBytes, &networkInfo)
 	if err != nil {
-		return nil, fmt.Errorf("failed to unmarshal network-pci map %w", err)
+		return nil, fmt.Errorf("failed to unmarshal network-info annotation: %w", err)
+	}
+
+	networkPciMap := map[string]string{}
+	for _, iface := range networkInfo.Interfaces {
+		if iface.DeviceInfo != nil && iface.DeviceInfo.Pci != nil && iface.DeviceInfo.Pci.PciAddress != "" {
+			networkPciMap[iface.Network] = iface.DeviceInfo.Pci.PciAddress
+		}
```

Comment:
Why is this needed?
Is it because the SR-IOV interfaces have not been filtered at the controller?

Reply:
Previously the networkPciMap annotation was in form of map[string]string and it could be used as is.
The new network-info annotation is in different form and require some processing to get the relevant info (network to pci-address mapping)

Reply:
The network binding interfaces should be treated in the relevant sidecar. So I agree the SRIOV interface should be filtered. But I don't see a reason to generalize the package.